### PR TITLE
Revert "rgw/auth: Fix the return code returned by AuthStrategy,"

### PR DESCRIPTION
**Pull request by pritha-srivastava**
This reverts commit 6f7ef8620e52be776ef633bb962a4309996529df.

the above fix introduced regression for ec2 auth when rgw_s3_auth_order has local before external.

fixes: https://tracker.ceph.com/issues/68393

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by pritha-srivastava** [comment]
@cbodley : s3tests changes are here: https://github.com/ceph/s3-tests/pull/602

**Comment by cbodley** [comment]
test results against https://github.com/ceph/s3-tests/pull/602 and https://github.com/ceph/ceph/pull/60122 in https://pulpito.ceph.com/cbodley-2024-11-20_14:25:57-rgw-wip-rgw-revert-auth-code-distro-default-smithi/

lots of red from known java issues, but both of the rgw/tempest jobs (`auth-order/local-external` and `external-local`) passed 🎉 

the new job to run the rgw/sts suite with reversed auth order (`auth-order/local-sts`) failed though. i added details to https://tracker.ceph.com/issues/69001 and pushed a candidate fix to the test in https://github.com/ceph/s3-tests/pull/604. if that doesn't work, we can omit that `auth-order/local-sts` test for now

**Comment by cbodley** [comment]
@pritha-srivastava would you mind pulling the two commits from https://github.com/ceph/ceph/pull/60122 into this pr so we can test/merge/backport together?

File: qa/rgw/s3tests-branch.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 overrides:
   s3tests:
-    force-branch: ceph-master
-    # git_remote: https://github.com/ceph/
+    force-branch: wip-rgw-change-err-code
+    git_remote: https://github.com/pritha-srivastava/
```
Comment (cbodley) [inline qa/rgw/s3tests-branch.yaml:4]: i went ahead and merged https://github.com/ceph/s3-tests/pull/602 and https://github.com/ceph/s3-tests/pull/604 to ceph-master, so we can drop this stuff for testing

**Comment by pritha-srivastava** [comment]
> @pritha-srivastava would you mind pulling the two commits from #60122 into this pr so we can test/merge/backport together?

@cbodley : done

File: qa/rgw/s3tests-branch.yaml (hunk 1 above)
Comment (pritha-srivastava) [reply qa/rgw/s3tests-branch.yaml]: done

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
jenkins test make check arm64

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
finally passed qa in https://pulpito.ceph.com/cbodley-2024-12-09_17:44:12-rgw-wip-68393-distro-default-smithi/ with rerun https://pulpito.ceph.com/cbodley-2024-12-10_14:47:37-rgw-wip-68393-distro-default-smithi/

**Pull request merged**